testplan/runnable/interactive/http.py

```diff
@@ -319,15 +319,18 @@ def should_run(curr_status):
         setting the state of a running test to PASSED) is not allowed - only
         the server may make those transitions. A BadRequest exception will be
         raised if the requested status is not valid.
+
+        TODO: from api design perspective, should_run should take a curr_status
```

Comment:
Could add a helper function `get_requested_status()` to handle extracting the new_status from the request JSON, then can call like:

```
if should_run(curr_status, get_requested_status()):
```

Reply:
i'm not addressing this in this PR